mautic/mautic
# Option to disable assets like PDF from Google and other search engines

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | 
| New feature? | y
| Automated tests included? |
| Related user documentation PR URL | https://github.com/mautic/documentation/pull/252
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | 
| Deprecations? | 

[//]: # ( Note that all new features should have a related user and/or developer documentation PR in their respective repositories. )

This bugfix/feature has been sponsored by @Webmecanik

[//]: # ( Required: )
#### Description:
PR added new option to disable search bots to crawl assets using the X-Robots-Tag HTTP header (more https://developers.google.com/search/reference/robots_meta_tag#using-the-x-robots-tag-http-header)

![image](https://user-images.githubusercontent.com/462477/35703633-0b63650c-079d-11e8-8f37-48d2a94d1ae4.png)

#### Steps to test this PR:
1. Apply PR
2.  Create local assets (PDF)
3. You can use Fetch as Google for websites https://support.google.com/webmasters/answer/6066468?hl=en for validation header. You should see X-Robots-Tag in header like this:
![image](https://user-images.githubusercontent.com/462477/35704744-beccc3ba-07a0-11e8-92e6-dafdc760e6ee.png)
4. Make sure that there will be no tag when the checkbox is turned off.

Another way how to test robots header is in inspection of your browser:

![image](https://user-images.githubusercontent.com/462477/37777260-613fadbc-2de7-11e8-9c60-3d0a072fe75f.png)
